File: apzkr-pzpi-21-2-volnykov-kyrylo/Task1-Server/AquaTrack/AquaTrack/Models/SensorData.cs
namespace AquaTrack.Models
{
    public enum SensorType
    {
        TemperatureSensor,
        WaterLevelSensor,
        LightingSensor,
        AciditySensor,
        MovementSensor,
        VideoSensor
    }

    public class SensorData
    {
        public int SensorDataId { get; set; }
        public SensorType SensorType { get; set; }
        public double SensorValue { get; set; }
        public DateTime? Timestamp { get; set; }
        public string? SensorStatus { get; set; }
        public string SensorIdentificator { get; set; }

        public List<ResearchReport> ResearchReports { get; set; }
    }
}

File: apzkr-pzpi-21-2-volnykov-kyrylo/Task4-MobileClient/AquaTrackMobile/app/src/main/java/com/example/aquatrackmobile/models/SensorData/SensorData.kt
package com.example.aquatrackmobile.models.SensorData

import com.example.aquatrackmobile.models.Aquarium.Aquarium
import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName

@Serializable
data class SensorDataResponse(
    val sensorData: SensorData
)

@Serializable
data class SensorData(
    val sensorDataId: Int,
    val sensorType: Int,
    val sensorValue: String,
    val timestamp: String,
    val sensorStatus: String,
    val sensorIdentificator: String
)

enum class SensorType(val id: Int) {
    @SerialName("0") TEMPERATURE(0),
    @SerialName("1") WATERLEVEL(1),
    @SerialName("2") LIGHTNING(2),
    @SerialName("3") ACIDITY(3),
    @SerialName("4") MOVEMENT(4),
    @SerialName("5") VIDEO(5);
}